ksqldb-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java

```diff
@@ -1474,9 +1474,13 @@ private static OptionalInt getLimit(final LimitClauseContext limitContext) {
   }
 
   private static Set<SourceName> getSources(final ParseTree parseTree) {
-    final SourceAccumulator accumulator = new SourceAccumulator();
-    accumulator.visit(parseTree);
-    return accumulator.getSources();
+    try {
+      final SourceAccumulator accumulator = new SourceAccumulator();
+      accumulator.visit(parseTree);
+      return accumulator.getSources();
+    } catch (StackOverflowError e) {
+      throw new KsqlException("Error processing statement: Statement is too large to parse.");
```

Comment:
nit: might make sense to include the GH ticket in the error message so that users can understand which part and what to do about it (namely, make the nested parenthesis less nested)

Reply:
I updated the error messages to suggest that nested expressions could be causing the error.